### ISSUE TITLE
How to add `data.x` property to graphs without vertex labels?

### ISSUE THREAD

**Issue by leftthomas** [rusty1s/pytorch_geometric#140]
I have read the paper **An End-to-End Deep Learning Architecture for Graph Classification**, and when I implement  the model, I found some datasets (such as **COLLAB**) have no `Node Labels` or `Node Attr. `.
So if I use `TUDataset` class to load the data, it will lack `data.x` property. The paper stated 

> For graphs without vertex labels, X can be defined as a column vector of normalized node degrees.

Then I tried to define a `Transform` class to implement this, here is my code:
```python
class Indegree(object):
    r"""Adds the globally normalized node degree to the node features.

    Args:
        cat (bool, optional): If set to :obj:`False`, all existing node
            features will be replaced. (default: :obj:`True`)
    """

    def __init__(self, norm=True, max_value=None, cat=True):
        self.norm = norm
        self.max = max_value
        self.cat = cat

    def __call__(self, data):
        col, x = data.edge_index[1], data.x
        deg = degree(col, data.num_nodes)

        if self.norm:
            deg = deg / (deg.max() if self.max is None else self.max)

        deg = deg.view(-1, 1)

        if x is not None and self.cat:
            x = x.view(-1, 1) if x.dim() == 1 else x
            data.x = torch.cat([x, deg.to(x.dtype)], dim=-1)
        else:
            data.x = deg

        return data

    def __repr__(self):
        return '{}(norm={}, max_value={})'.format(self.__class__.__name__, self.norm, self.max)
```
Finally I invoke it by `TUDataset('data', 'COLLAB', pre_transform=Indegree())`, is this correct?

**Comment by rusty1s** [comment]
Looks good to me!

**Comment by leftthomas** [comment]
@rusty1s thanks for your reply
